picard/coverart/imagelist.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+# Copyright (C) 2017 Antonio Larrosa <alarrosa@suse.com>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+
+def get_image_type(image):
+    return image.types_as_string()
+
+
+class ImageList(list):
+
+    def __init__(self):
+        super(ImageList, self).__init__()
+
+    def __eq__(self, other):
+        return sorted(self, key=get_image_type) == sorted(other, key=get_image_type)
+
+    def __getslice__(self, i, j):
```

Comment:
Yes, it's deprecated but it has to be used in this case: (from https://docs.python.org/2.7/reference/datamodel.html#object.__getslice__ ) "Deprecated since version 2.0: Support slice objects as parameters to the __getitem__() method. (However, built-in types in CPython currently still implement __getslice__(). Therefore, you have to override it in derived classes when implementing slicing.)" . So in general it shouldn't be used but for classes derived from lists and dicts, __getslice__ has to be used (only in Python 2).